runtime/bcutil/ClassFileOracle.hpp

```diff
@@ -921,6 +923,17 @@ class NameAndTypeIterator
 	bool annotationRefersDoubleSlotEntry() const { return _annotationRefersDoubleSlotEntry; }
 	bool isInnerClass() const { return _isInnerClass; }
 
+	U_8 constantDynamicType(U_16 cpIndex) const
+	{
+		J9CfrConstantPoolInfo* nas = &_classFile->constantPool[_classFile->constantPool[cpIndex].slot2];
+		J9CfrConstantPoolInfo* signature = &_classFile->constantPool[nas->slot2];
+		if ('D' == signature->bytes[signature->slot1 - 1]) {
+			return JBldc2dw;
+		} else {
```

Comment:
If this is only intended to be used by `ldc2*` bytecodes, then please make this explicitly check the two cases and assert otherwise.

Reply:
fixed